### ISSUE TITLE
Error when parsing single-quoted URL strings 

### ISSUE THREAD

**Issue by mre** [robinst/linkify#20]
There is a bug with single-quoted URLs.

Input:
```
'https://example.org''
```

Expected:
```
https://example.org
```

Actually extracted link:
```
https://example.org''
```

Downstream issue in lychee: https://github.com/lycheeverse/lychee/issues/230

**Comment by robinst** [comment]
Hmm this is an interesting one. Note that GitHub's own autolinking does the same thing: https://www.google-analytics.com/analytics.js','ga

A couple of different ideas to fix this:

1. Check if there's a quote *before* the scheme, and if yes, stop the URL once we encounter one of those quotes. That would mean `'https://example.org''` would not include the quotes, but something like `'https://example.com, https://example.org','foo` would still include the quotes in the second URL. Maybe that's good enough? It would fix your specific example.
2. If the first character after the first `'` is some kind of delimiter like `,` or `;` or `.` (what else?), stop the URL. That would even get the `'https://example.com, https://example.org','foo` example right, but maybe it's too restrictive?

Would love to hear your thoughts. Do you have any test cases for lychee where we could see if one of the options works better?

**Comment by mre** [comment]
> Note that GitHub's own autolinking does the same thing

We are in good company. 😆 

Option 1 sounds like a good idea to me. I'd say that `'https://example.com, https://example.org','foo` would indicate a syntax error anyway; so perhaps we can ignore that case (for now). There are quite a few [quote characters](https://www.cl.cam.ac.uk/~mgk25/ucs/quotes.html), though. Also, how would we handle escaped quotes (e.g. `\'https://example.org'`? My suggestion is to create a static array of "quote characters" (or use a crate if we find one) and not care about edge-cases too much right now.

Option 2 sounds more complicated because we could easily miss delimiters or edge-cases. For instance `https://example.org'()` is a valid URL according to [RFC 1738](http://www.faqs.org/rfcs/rfc1738.html) (Also see this [StackOverflow answer](https://stackoverflow.com/a/3867794/270334))  &mdash; even though the special characters are not encoded. We might run into other issues down the road.

>  Do you have any test cases for lychee where we could see if one of the options works better?

Apart from the Google Analytics link above, no. I'll keep an eye out for more.

**Comment by mre** [comment]
I've updated the PR at #21 according to option 1.

**Comment by robinst** [comment]
Released in 0.7.0 🎉: https://github.com/robinst/linkify/blob/main/CHANGELOG.md#070---2021-05-18

**Comment by mre** [comment]
Yay. Thanks!
